lib/fizzy/execute.cpp

```diff
@@ -324,6 +326,19 @@ inline DstT extend(SrcT in) noexcept
         return DstT{in};
 }
 
+template <typename SrcT, typename DstT>
+inline bool trunc(OperandStack& stack) noexcept
+{
+    using boundaries = trunc_boundaries<SrcT, DstT>;
```

Comment:
Why would it not work with `static_cast<SrcT>(std::numeric_limits<Dst>::min())` / `static_cast<SrcT>(std::numeric_limits<Dst>::max())` ?

Reply:
E.g.
- `-0.0 -> 0u`
- `-0.999 -> 0u`
- `-1.0 -> undefined`

Reply:
I think it is rather difficult to have generic formula for upper/lower boundaries. However, it is possible to compress these (e.g. for unsigned type these are `-1.0` and `2^N`. But I decided to leave these expanded at least until we have confirmation that all spectests are passed.

Having separate file for it is not ideal, but I wanted to check the values against math formulas used in unit tests.

Reply:
I think it looks fine now, I like that it's quite simple. The only thing is that  bounds look like magic values, they are not defined in spec. It needs a comment to explain where they come from.

Reply:
Added comments.